### ISSUE TITLE
Request for Documentation: Deployment Guides for FrankenPHP on Various Platforms

### ISSUE THREAD

**Issue by Trusted97** [dunglas/frankenphp#220]
Hello @dunglas ,

I hope this message finds you well. I have been thoroughly enjoying the capabilities and features offered by FrankenPHP for PHP application deployment. It's an incredible tool for enhancing PHP applications, and I appreciate the effort you've put into developing it.

I was wondering if it would be possible to include comprehensive deployment guides in the FrankenPHP documentation. Specifically, I believe it would greatly benefit users like myself if there were step-by-step guides on deploying FrankenPHP-powered applications on popular cloud platforms such as DigitalOcean, AWS, Heroku, and others.

These guides would ideally cover the following aspects:

1. Setting up a FrankenPHP environment on the chosen cloud platform.
2. Configuring FrankenPHP to work optimally with the platform's infrastructure.
3. Best practices for scaling and managing FrankenPHP applications.
4. Tips for optimizing performance and security in a cloud environment.

I believe that having such guides readily available in the documentation would not only make it easier for existing users to deploy FrankenPHP applications but also attract new users looking for a powerful and efficient way to host their PHP applications on these cloud platforms.

If you could consider adding these deployment guides to the documentation, it would be greatly appreciated and would undoubtedly enhance the overall user experience.

Thank you for your time and consideration. I look forward to seeing FrankenPHP continue to evolve and thrive.

**Comment by luisalvarado** [comment]
Yes, was gonna ask this. Took me a while on a local ubuntu to figure the following out:

`sudo /home/luis/web/frankenphp-linux-x86_64 start config /etc/caddy/Caddyfile `

For example in my case I mainly use caddy + php-fpm + mysql. I use other stuff but for this particular case I install caddy via their PPA, php-fpm from the repo, same for mysql. Which means, I do not use symfony, nor docker.

How for example

1. Can I use frankenphp as a service? (With or without docker, in my case without it)
2. How to configure the internal Caddy server file (Already solved with the investigation I did) by calling the start and config parameters. But it was not clear.
3. How to use mercure (without docker, composer, etc..) Like is it internally inside frankenphp or how is it configure). I have no clue how to configure it or even test it at this point.

Other questions about a step by step guide to understand this would be much appreciated. In the mean time I am guessing stuff by reading he mercure documentation. for example, how to update frankenphp when PHP 8.3 comes out.

**Comment by sneycampos** [comment]
> Yes, was gonna ask this. Took me a while on a local ubuntu to figure the following out:
> 
> 
> 
> `sudo /home/luis/web/frankenphp-linux-x86_64 start config /etc/caddy/Caddyfile `
> 
> 
> 
> For example in my case I mainly use caddy + php-fpm + mysql. I use other stuff but for this particular case I install caddy via their PPA, php-fpm from the repo, same for mysql. Which means, I do not use symfony, nor docker.
> 
> 
> 
> How for example
> 
> 
> 
> 1. Can I use frankenphp as a service? (With or without docker, in my case without it)
> 
> 2. How to configure the internal Caddy server file (Already solved with the investigation I did) by calling the start and config parameters. But it was not clear.
> 
> 3. How to use mercure (without docker, composer, etc..) Like is it internally inside frankenphp or how is it configure). I have no clue how to configure it or even test it at this point.
> 
> 
> 
> Other questions about a step by step guide to understand this would be much appreciated. In the mean time I am guessing stuff by reading he mercure documentation. for example, how to update frankenphp when PHP 8.3 comes out.

You can use it as standalone without docker, did you try or did i understand wrong?

**Comment by luisalvarado** [comment]
@sneycampos that's the problem. There is no documentation about how to do it. I went from the official page to the github and back. Was not able to find someone step by step to at least have a base to use frankenphp. That is what I am referring to with all the points.

**Comment by dunglas** [comment]
Good idea! Having such guides would definitely serve the project. Pull requests welcome!

Also, providing packages for popular distributions (Debian, Homebrew...) would help a lot.

**Comment by Trusted97** [comment]
Hello @dunglas,
Do you know if there's any article or older documentation I could refer to as a starting point for writing the documentation?

**Comment by francislavoie** [comment]
@Trusted97 running FrankenPHP is effectively just "running Caddy", so you should probably refer to the Caddy docs to answer your questions. For example: https://caddyserver.com/docs/running

**Comment by luisalvarado** [comment]
> @Trusted97 running FrankenPHP is effectively just "running Caddy", so you should probably refer to the Caddy docs to answer your questions. For example: https://caddyserver.com/docs/running

Hi Francis, awesome new documentation thank you. Would you help me i knowing, where should I got if I want to leave feedback about how to improve the documentation for caddy. I started with caddy a bunch of years ago and went over a lot of documentation iterations. This last one is looking great but in case I would want to help with quality, how may I do that. Thank you for all the work you do as always.

**Comment by francislavoie** [comment]
Our docs are at https://github.com/caddyserver/website

That page isn't new, it's existed for over two years https://github.com/caddyserver/website/commit/5f09f05e0825f0835691788699d32698faa55a54

**Comment by luisalvarado** [comment]
> The Caddy website source is at https://github.com/caddyserver/website
> 
> That page isn't new, it's existed for over two years [caddyserver/website@5f09f05](https://github.com/caddyserver/website/commit/5f09f05e0825f0835691788699d32698faa55a54)

That is weird because the last times I tried the tutorial it always took me to a different one. The last time meaning about 6 months ago more or less.

**Comment by francislavoie** [comment]
It's under "Articles", not "Tutorials", in the sidebar.

**Comment by Trusted97** [comment]
Hi to everyone,
I've made this [PR](https://github.com/dunglas/frankenphp/pull/261) for add Digital Ocean Documentation on FrankenPHP
